[Bug]: Unittest TestConnectionManager/rootCoord not stable

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:aa84124
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

Unit test case `TestConnectionManager/rootCoord` fails randomly.

### Expected Behavior

Unit test shall pass stably.

### Steps To Reproduce

_No response_

### Milvus Log

```
2023-01-30T06:29:48.1095016Z --- FAIL: TestConnectionManager (10.62s)
2023-01-30T06:29:48.1095439Z     --- FAIL: TestConnectionManager/rootCoord (10.01s)
2023-01-30T06:29:48.1095842Z     --- PASS: TestConnectionManager/queryCoord (0.11s)
2023-01-30T06:29:48.1096233Z     --- PASS: TestConnectionManager/dataCoord (0.11s)
2023-01-30T06:29:48.1096627Z     --- PASS: TestConnectionManager/queryNode (0.11s)
2023-01-30T06:29:48.1096994Z     --- PASS: TestConnectionManager/dataNode (0.11s)
```

### Anything else?

Action link:  https://github.com/milvus-io/milvus/actions/runs/4041077778/jobs/6947304280

--- a/internal/distributed/connection_manager_test.go
+++ b/internal/distributed/connection_manager_test.go
@@ -40,8 +40,26 @@ import (
 	"google.golang.org/grpc"
 )
 
-func TestConnectionManager(t *testing.T) {
+func TestMain(t *testing.M) {
+	// init embed etcd
+	embedetcdServer, tempDir, err := etcd.StartTestEmbedEtcdServer()
+	if err != nil {
+		log.Fatal("failed to start embed etcd server for unittest", zap.Error(err))
+	}
+
+	defer os.RemoveAll(tempDir)
+	defer embedetcdServer.Server.Stop()
+
+	addrs := etcd.GetEmbedEtcdEndpoints(embedetcdServer)
+	// setup env for etcd endpoint
+	os.Setenv("etcd.endpoints", strings.Join(addrs, ","))
+
 	paramtable.Init()
+
+	os.Exit(t.Run())
+}
+
+func TestConnectionManager(t *testing.T) {
 	ctx := context.Background()
 
 	session := initSession(ctx)

